zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/store/ExporterBatchWriter.java

```diff
@@ -73,6 +73,7 @@ public void flush(final BatchRequest batchRequest) throws PersistenceException {
       final ExporterEntity entity = entityAndHandler.entity();
       for (final var handler : entityAndHandler.handlers()) {
         handler.flush(entity, batchRequest);
+        handler.onError(batchRequest);
       }
     }
     batchRequest.execute();
```

Comment:
We pass in here a lambda or function - then based on the parameters like id, index, etc. we can find the handler and call it respectively.